classes/meson.oeclass

```diff
@@ -0,0 +1,123 @@
+# build rules for meson
+# Needed for latest versions of gnome packages. 
+
+# remember to install
+# python >= 3.5
+# pip3 install meson
+# ninja-build
+
+def meson_array(d,var):
+    items = d.get(var).split()
+    return repr(items[0] if len(items) == 1 else items)
+
+# Map our ARCH values to what Meson expects:
+# http://mesonbuild.com/Reference-tables.html#cpu-families
+def meson_cpu_family(d,var):
+    import re
+    arch = d.getVar(var)
+    if arch == 'powerpc':
+        return 'ppc'
+    elif arch == 'powerpc64':
+        return 'ppc64'
+    elif arch == 'mipsel':
+        return 'mips'
+    elif arch == 'mips64el':
+        return 'mips64'
+    elif re.match(r"i[3-6]86", arch):
+        return "x86"
+    else:
+        return arch
+
+def meson_endian(d,prefix):
+    endian = d.getVar(prefix + "_ENDIAN")
+    if endian == 'l':
+        return "little"
+    elif endian == 'b':
+        return "big"
+    else:
+        bb.fatal("Cannot determine endianism for %s" % (prefix))
+
+OE_MESON_CROSSFILE = "${STAGE_DIR}/cross/meson.cross"
```

Comment:
This is bad.  You are not supposed to write to `${STAGE_DIR}`.  Please write this to `${B}` instead.